File: backend/app/schemas/memo.py
from __future__ import annotations

from pydantic import BaseModel, Field


class MemoRequest(BaseModel):
    ticker: str = Field(min_length=1, max_length=16)


class MemoSection(BaseModel):
    title: str
    body: str


class MemoResponse(BaseModel):
    id: int | None = None
    ticker: str
    company: str
    rating: str            # Buy | Hold | Sell
    conviction: str        # High | Moderate | Low
    target_price: float | None
    current_price: float | None
    upside_pct: float | None
    health_score: float
    sentiment: str
    sections: list[MemoSection]
    created_at: str | None = None

File: web/src/lib/memo.ts
export interface MemoSection { title: string; body: string; }
export interface MemoResponse {
  id: number;
  ticker: string;
  company: string;
  rating: "Buy" | "Hold" | "Sell";
  conviction: string;
  target_price: number | null;
  current_price: number | null;
  upside_pct: number | null;
  health_score: number;
  sentiment: string;
  sections: MemoSection[];
  created_at: string;
}
